src/hw1/task1.py

```diff
@@ -0,0 +1,80 @@
+from math import sqrt
+from copy import deepcopy
+from typing import Union, List, Tuple
+
+Num = Union[float, int]
+Vector = List[Num]
+Matrix = List[List[Num]]
```

Comment:
Is there any particular reason why you could not say `Matrix = List[Vector]`? Not that it's wrong, but might be easier to understand this way.